python/felis/datamodel.py

```diff
@@ -462,3 +465,27 @@ def require_description(cls, rd: bool = True) -> None:
     def is_description_required(cls) -> bool:
         """Return whether a description is required for all objects."""
         return cls.ValidationConfig._require_description
+
+    @classmethod
+    def from_yaml_path(cls, yaml_path: str) -> Schema:
+        with open(yaml_path, "r") as yaml_file:
+            yaml_data = yaml.load(yaml_file, Loader=yaml.SafeLoader)
+        schema = cls.model_validate(yaml_data)
+        return schema
+
+    @classmethod
+    def from_yaml_file(cls, yaml_file: TextIOBase) -> Schema:
+        yaml_data = yaml.load(yaml_file, Loader=yaml.SafeLoader)
+        schema = cls.model_validate(yaml_data)
+        return schema
+
+    @classmethod
+    def from_yaml_gh(
+        cls, schema_name: str, repo: str, branch: str = "main", dir_append: str | None = None
+    ) -> Schema:
+        gh_url = f"https://raw.githubusercontent.com/{repo}/{branch}/{dir_append}/{schema_name}.yaml"
```

Comment:
Or change `from_yaml_path` to support `ResourcePath` and then you don't need the rest of this method because that would work for any URIs.

Reply:
I'm not sure what you mean by `ResourcePath` here. I can't find a class with this name in the standard Python library.

Reply:
@timj I will rework these utility functions to be less duplicative with better naming so that the purpose of each is clear.

Reply:
I'm going to drop these functions from the PR and put them on a separate ticket.